### PR TITLE
feat(deploy): Update vespa deployment script to handle model changes

### PR DESCRIPTION
**Pull request by junaid-shirur**
- Update vespa deployment script to handle model change via .env value 
`EMBEDDING_MODEL=`

### CODE REVIEW

File: server/vespa/deploy.sh [hunk 1]
```diff
@@ -1,12 +1,34 @@
 #!/bin/sh
 set -e
+source ../.env
```
Comment (zereraz) [inline server/vespa/deploy.sh]: remove this, assume the env variables will be already present

File: server/vespa/replaceDIMS.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        modelDimension
+      );
+    } catch (error) {
+      console.error(`Failed to process ${p}:`, error);
+    }
+  });
+};
+
+const args = process.argv.slice(2)
+const getDotSdpaths = (directory: string) => {
```
Comment (zereraz) [inline server/vespa/replaceDIMS.ts]: name it `getSchemaDefinitionPaths`

File: server/vespa/replaceDIMS.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    return fs.readdirSync(directory)
+        .filter(file => path.extname(file) === ".sd")
+        .map(file => path.join(directory, file));
+};
+
+const paths = getDotSdpaths("./schemas");
+
+replaceModelDIMS(paths, parseInt(args[0]));
```
Comment (zereraz) [inline server/vespa/replaceDIMS.ts]: check the value of the dims we get, if undefined or null we throw error that dim's has to exist

File: server/vespa/deploy.sh (hunk 1 above)
Comment (junaid-shirur) [reply server/vespa/deploy.sh]: we require this, otherwise it will not take updated `.env` values

Comment (zereraz) [reply server/vespa/deploy.sh]: that is up to the way we deploy, assume env will be present when the script runs

**Pull request merged**